### ISSUE TITLE
Support for AdamW optimizer in ExecuTorch

### ISSUE THREAD

**Issue by ChhaviMehta17** [pytorch/executorch#18766]
As of now, ExecuTorch supports SGD as the only optimizer, with Adam listed as planned.

Given that AdamW is the default optimizer used in most modern PyTorch training workflows, is AdamW support also expected as part of the ExecuTorch optimizer roadmap?

Are there any known design or semantic considerations in existing AdamW implementations (e.g., weight decay behavior or optimizer state handling) that influence how AdamW support is approached in ExecuTorch?

Thanks!

cc @JacobSzwejbka

**Comment by SS-JIA** [comment]
Assigning to @JacobSzwejbka since this appears to be related to training.

**Comment by JacobSzwejbka** [comment]
I think its just a matter of porting it over. I dont think there is any fundamental blocker or anything.

**Comment by BryanBradfo** [comment]
Hi @JacobSzwejbka, I'd love to take this on! I will look into porting the AdamW optimizer implementation into ExecuTorch by using the existing SGD implementation as a reference

**Comment by psiddh** [comment]
@BryanBradfo can you complete EASY CLA (CI Failure) please ?
